android/trifle/src/main/java/app/cash/trifle/KeyHandle.kt

```diff
@@ -35,6 +37,28 @@ class KeyHandle internal constructor(private val alias: String) {
     }
   }
 
+  internal val keyPair: KeyPair by lazy {
+    val ks: KeyStore = KeyStore.getInstance("AndroidKeyStore").apply {
+      load(null)
+    }
+
+    // Throw an illegal state exception if we can't get hold of the proper key material. This
+    // *should never happen* since the only way to obtain a KeyHandle is to deserialize one, which
+    // should have already checked for this, or to generate a new one.
+    var exception: IllegalStateException? = null
+    val exceptionMessage =
+      "Android KeyStore does not contain a keypair corresponding to the $alias alias"
+    try {
+      val entry: KeyStore.Entry = ks.getEntry(alias, null)
+      if (entry is KeyStore.PrivateKeyEntry) {
+        KeyPair(entry.certificate.publicKey, entry.privateKey)
+      }
+    } catch (e: Exception) {
+      exception = IllegalStateException(exceptionMessage, e)
```

Comment:
mmm can we just rethrow the illegalstateexception here instead?

Reply:
done